### PR TITLE
Add support for :only keyword and reset config between test scenarios

### LINKED ISSUE #8: Monitor Files for Changes
Should we handle file changes when using the File loader? Would be kill the application configured for that key and let the supervisor bring it back up?

### PR DESCRIPTION
**Pull request by Betree**
Fixed all my warnings for unused configs with umbrella app. 

Fix #8 + #10

### CODE REVIEW

File: lib/loaders/file.ex [hunk 1]
```diff
@@ -28,14 +28,15 @@ defmodule Weave.Loaders.File do
   @spec load_configuration(atom()) :: list()
   @spec get_configured_directories() :: list() | binary() | :weave_no_directory
   @spec load_configuration(binary(), binary(), atom()) :: :ok
-  @spec load_configuration_from_directory(binary(), atom()) :: :ok
+  @spec load_configuration_from_directory(binary(), atom(), binary()) :: :ok
```
Comment (rawkode) [inline lib/loaders/file.ex]: Shouldn't this be a `list()` ?

Comment (Betree) [reply lib/loaders/file.ex]: Yup you're right ! Commit incomming

**Comment by Betree** [comment]
Fixed with 6463b0b 😺

**Comment by Luciam91** [comment]
Merged to the Gitlab canonical repo :) https://gitlab.com/gt8/open-source/elixir/weave/commit/72087700a39d90a76b284280083eb123dc5fa8a7

Thanks @Betree

**Comment by Betree** [comment]
Great :)

Simple curiosity @Luciam91, why hosting the main repo on Gitlab ?

**Comment by Luciam91** [comment]
@Betree We host on Gitlab due to hosting the rest of our projects on there as well, this also allows us to run our builds on our own Gitlab runners rather than using something like Travis CI.

@rawkode can you potentially provide any additional details

**Comment by rawkode** [comment]
Hey @Betree,

Why GitLab? @Luciam91 is correct, but I'll expand a little:

1. GitLab CI is Awesome

We love GitLib CI. We add a simple `yml` file to our repository and CI is enabled. As @Luciam91 also said, we can provide our own runners for the jobs, so we don't rely on shared hardware.

2. Grouping / Naming / Structure

Just look at how things are setup - we can group our OSS together, further with sub-groups for languages. This structure is clean and simply to understand, rather than GitHub's massive list of projects

3. Transparency

This is the most important one. GitLab has a public, transparent, roadmap and is a forerunner and thoughtleader in how an OSS company should look. Yes, they suffer a lot of downtime and it can get frustrating - but the transparency in roadmap, feature requests and their very public post-mortems give me the confidence I need to support them

We wouldn't even mirror to GitHub if we didn't need to, but sadly a lot of people disregard OSS projects if they don't live here too.

**Comment by Betree** [comment]
Thanks! That's an amazing insight. The (3.) really make sense.

I was curious cause I'm currently doing the opposite move: moving from Gitlab to Github. The main reason for this is that I'm open-sourcing all the code. Travis / Coveralls provide great services for free when you work on public projects without the need to allocate custom resources (runners).
And like you said, a lot of people disregard OSS projects if they don't live on GIthub.

Strange to think that open-source, which is by nature something extremely decentralized, suffer from centralization.

I'll contribute directly on Gitlab next time ;)

**Pull request closed without merging**